src/predictor.js

```diff
@@ -51,6 +51,9 @@ export function applyPredictor(block, predictor, width, height, bitsPerSample) {
   const stride = bitsPerSample.length;
 
   for (let i = 0; i < height; ++i) {
+    // Last strip will be truncated if height % stripHeight != 0
+    if (i * stride * width * bytesPerSample >= block.byteLength)
```

Comment:
Just brainstorming, but would it be possible to move the logic from line 55 outside the for loop and instead use:
```js
const imax = Math.min(height, block.byteLength / (stride * width * bytesPerSample));
for (let i = 0; i < imax; ++i) {
```
That way we could run that operation once instead of for each iteration.  Happy to be corrected if I'm wrong :-)

Reply:
Oops. I commented before I reviewed the new PR which takes the same approach: https://github.com/geotiffjs/geotiff.js/pull/90/files#diff-984eadc2655393cb6c8d068e0a107347R54